go-controller/pkg/libovsdb/ops/switch.go

```diff
@@ -286,15 +286,15 @@ func GetLogicalSwitchPort(nbClient libovsdbclient.Client, lsp *nbdb.LogicalSwitc
 	return found[0], nil
 }
 
-func createOrUpdateLogicalSwitchPortsOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, sw *nbdb.LogicalSwitch, createSwitch bool, lsps ...*nbdb.LogicalSwitchPort) ([]libovsdb.Operation, error) {
+func createOrUpdateLogicalSwitchPortsOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, sw *nbdb.LogicalSwitch, createSwitch bool, customFields []nbdb.LSPCustomField, lsps ...*nbdb.LogicalSwitchPort) ([]libovsdb.Operation, error) {
 	originalPorts := sw.Ports
 	sw.Ports = make([]string, 0, len(lsps))
 	opModels := make([]operationModel, 0, len(lsps)+1)
 	for i := range lsps {
 		lsp := lsps[i]
 		opModel := operationModel{
 			Model:          lsp,
-			OnModelUpdates: getAllUpdatableFields(lsp),
+			OnModelUpdates: append(getDefaultUpdatableFields(lsp), getCustomUpdatableFields(lsp, customFields)...),
```

Comment:
I don't think we should update anything else than what we have been told to update.

Reply:
tbh that's not very specific enough to me, but if I understand you correctly, then there should be no "default" values for `createOrUpdateLogicalSwitchPortsOps`. 
Instead, there should only be a specific list of fields that this function should be told to update. 
We still want to keep the defaultFields inside as a private function right? My reasoning is - no need to make that function public, exposing implementation detail outside the library

So this means I should change the following:
1. The public `CreateOrUpdateLogicalSwitchPortsOnSwitchOpsWithCustomFields` function will create a list of the enum fields. it will be comprised by the default fields `getDefaultUpdatableFields` and the custom fields, in case needed. 

2 . `createOrUpdateLogicalSwitchPortsOps` should only get the fields to update (a.k.a the enum list)

I think that's a good compromise between the need to use custom fields (that need to be set from outside) and the default ones. tell me what you think.

Reply:
It's simpler than that. 

Just check if any field was provided. If fields where provided, use only those. If not, use the default. We follow this same patter elsewhere.

Reply:
ACK np, but keep in mind that I will need to expose the `getDefaultUpdatableFields` for one of the scenarios.